### PR TITLE
Add a more customizable navbar menu (#4442)

### PR DESCRIPTION
**Pull request by vlo-rte**
- In release note :
  -  In chapter : Features
  -  Text : #4442 : Add a more customizable navigation bar menu

### CODE REVIEW

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Comment by freddidierRTE** [comment]
with following config : 
```
    {
      "id": "menu1",
      "label": "title.single",
      "entries": [
        {
          "opfabCoreMenuId": "admin"
        },
        {
          "opfabCoreMenuId": "logging"
        }
      ]
    },
```
image

File: ui/main/src/app/business/services/menu.service.ts [hunk 1]
```diff
@@ -23,35 +23,80 @@ export class MenuService {
...
+        const topRightMenus = this.configService.getTopRightMenus();
+
+        if (topRightMenus) {
+            return topRightMenus
+                .filter((coreMenuConfig: CoreMenuConfig) => {
+                    return coreMenuConfig.visible && this.isMenuVisibleForUserGroups(coreMenuConfig);
+                })
+                .map((coreMenuConfig: CoreMenuConfig) => coreMenuConfig.opfabCoreMenuId);
+        } else {
+            console.log('No topRightMenus property set in ui-menu.json');
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/business/services/menu.service.ts:87]: Done.

File: src/docs/asciidoc/resources/migration_guide_to_3.15.adoc [hunk 2]
```diff
@@ -9,8 +9,25 @@
...
+
+* `navigationBar` : contains the navigation bar mixing core menus and custom menus
+* `topRightIconMenus` : contains only the two menu icons `agenda` and `usercard` on the top right of the screen
+* `topRightMenus` : contains the core menus you want to see when you click the user, on the top right of the screen
+* `locales` : contains the translations for the custom menus
+
+NOTE: With the new structure of the file, it is now possible to mix core menus and custom menus in the navigation bar.
+
+You can find more information and a full example in the documentation :
+https://opfab.github.io/documentation/current/reference_doc/#menu_entries
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply src/docs/asciidoc/resources/migration_guide_to_3.15.adoc]: Done.

File: src/docs/asciidoc/reference_doc/ui_customization.adoc [hunk 3]
```diff
@@ -94,17 +113,17 @@ for each core menu:
 
 See `/config/docker/ui-menu.json` for an example containing all core menus.
 
-IMPORTANT: This property only manages menu visibility, not resource access rights. For example, if visibility of the
-Administration menu is granted to groups other than the `ADMIN` group, users will be able to see the administration
-screen but not to perform administration tasks, as they require administrator privileges.
+IMPORTANT: This property only manages menu visibility, not resource access rights. For example, if visibility of
+the Administration menu is granted to the groups other than the `ADMIN` group, users will be able to see the
+administration screen but not to perform administration tasks, as they require administrator privileges.
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply src/docs/asciidoc/reference_doc/ui_customization.adoc:119]: Done.

**Pull request merged**